README.md

```diff
@@ -4,40 +4,112 @@ Focal point for gathering and analyzing biomedical evidence as a graph.
 
 ![GAIA](https://github.com/bmeg/gaia/blob/master/resources/gaia.jpg)
 
-## motivation
+## idea
 
-Gaia is an event, query and analysis engine built on the Tinkerpop graph db interface. It has three primary concepts: Schemas, Facets and Agents.
+Gaia tracks a set of keys representing files to be computed, and the network of dependent processes that compute them. At its core is some Data Store that holds these keys, which can begin life seeded with prior information. At each cycle, Gaia calculates which keys are missing and which processes are able to be run given the set of keys available, and matches these up to select a process to run whose inputs are all available and one or more outputs are missing. Once these missing keys are computed, the cycle is run again and again until either all keys are computed or no more processes can be run.
 
-### schemas
+## prerequisites
 
-Every Gaia instance has a schema.
+In order to run Gaia, you must have access to a Funnel server and whatever Kafka cluster Funnel is using.
```

Comment:
Is this adding a hard dependency on Funnel? Shouldn't the graph database (Gaia) and the data processing system (Gaia-flow?) live separately? Can't this data processing system be a plugin or complementary service?

Reply:
It is adding a hard dependency on Funnel, but since Funnel is an abstraction itself I am coding to what I believe Funnel will become instead of what it is now.

That said, it would be easy to make an adapter for other systems. I was under the impression however that Funnel *is* the adapter : )

The graph database is currently in Ophion, which Gaia has a dependency on, so they are currently living separately. I think you have been mislead in these last few issues by the current incarnation of Gaia living in the `flow` branch (what was the previous Gaia was the graph db, but no longer).

Reply:
> (what was the previous Gaia was the graph db, but no longer)

I think this is the key. I had the incorrect mapping of name to concept. Up until now, my understanding was:

- Ophion = query language
- Gaia = database
- Gaia-flow = data processing

Sounds like instead:
- Ophion = query language + database
- Gaia = data processing

Is that right? My question about making a hard dependency on Funnel was in the worldview of the dependency being made to a graph database (not a data processing system).